bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeAirDevice.java

```diff
@@ -0,0 +1,530 @@
+            List<Integer> valList = Arrays.asList(statusResponseGson.get().packJson.dat);
+            int valueArrayposition = colList.indexOf(valueName);
+            if (valueArrayposition != -1) {
+                // get the Corresponding value
+                Integer value = valList.get(valueArrayposition);
+                return value;
+            }
+        }
+
+        return -1;
+    }
+
+    public boolean isStatusAvailable() {
+        return statusResponseGson.isPresent();
+    }
+
+    public boolean hasStatusValChanged(String valueName) throws GreeException {
+        if (!prevStatusResponsePackGson.isPresent()) {
+            return true; // update value if there is no previous one
+        }
+        // Find the valueName in the Current Status object
+        List<String> currcolList = Arrays.asList(statusResponseGson.get().packJson.cols);
+        List<Integer> currvalList = Arrays.asList(statusResponseGson.get().packJson.dat);
+        int currvalueArrayposition = currcolList.indexOf(valueName);
+        if (currvalueArrayposition == -1) {
+            throw new GreeException("Unable to decode device status");
+        }
+
+        // Find the valueName in the Previous Status object
+        List<String> prevcolList = Arrays.asList(prevStatusResponsePackGson.get().cols);
+        List<Integer> prevvalList = Arrays.asList(prevStatusResponsePackGson.get().dat);
+        int prevvalueArrayposition = prevcolList.indexOf(valueName);
+        if (prevvalueArrayposition == -1) {
+            throw new GreeException("Unable to get status value");
+        }
+
+        // Finally Compare the values
+        return currvalList.get(currvalueArrayposition) != prevvalList.get(prevvalueArrayposition);
+    }
+
+    protected void executeCommand(DatagramSocket clientSocket, Map<String, Integer> parameters) throws GreeException {
+        // Only allow this to happen if this device has been bound
+        if (!getIsBound()) {
+            throw new GreeException("Device is not bound!");
+        }
+
+        try {
+            // Convert the parameter map values to arrays
+            String[] keyArray = parameters.keySet().toArray(new String[0]);
+            Integer[] valueArray = parameters.values().toArray(new Integer[0]);
+
+            // Prep the Command Request pack
+            GreeExecuteCommandPackDTO execCmdPackGson = new GreeExecuteCommandPackDTO();
+            execCmdPackGson.opt = keyArray;
+            execCmdPackGson.p = valueArray;
+            execCmdPackGson.t = GREE_CMDT_CMD;
+            String execCmdPackStr = gson.toJson(execCmdPackGson);
+
+            // Now encrypt and send the Command Request pack
+            String encryptedCommandReqPacket = GreeCryptoUtil.encryptPack(getKey().getBytes(), execCmdPackStr);
+            DatagramPacket sendPacket = createPackRequest(0,
+                    new String(encryptedCommandReqPacket.getBytes(), StandardCharsets.UTF_8));
```

Comment:
Why are you converting a string -> byte[] -> string?
```suggestion
            DatagramPacket sendPacket = createPackRequest(0, encryptedCommandReqPacket);
```

Reply:
changed